clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -105,6 +107,22 @@ public void disconnected(String id, long now) {
         nodeState.lastConnectAttemptMs = now;
     }
 
+    public void authFailed(String id, long now) {
```

Comment:
Nit: I think we should replace `authFailed` with `authenticationFailed` everywhere to avoid ambiguity (`authFailed` could be misinterpreted as `authorization failed`).